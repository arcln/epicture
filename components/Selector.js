import React from 'react';
import {View, Text, Platform} from 'react-native';
import OptionsMenu from 'react-native-options-menu';
import {Icon} from 'expo';
import Colors from '../constants/Colors';

export default class Selector extends React.Component {

  state = {
    value: this.props.default || '',
  }

  itemPressed = async value => {
    await this.setState({value});
    this.props.onChange && this.props.onChange(value);
  };

  render() {
    console.log(Array(this.props.options.length))
    return (
      <OptionsMenu
        options={[...this.props.options, 'Cancel']}
        customButton={(
          <View style={{flexDirection: 'row'}}>
            <Icon.Ionicons
              name={Platform.OS === 'ios' ? 'ios-arrow-down' : 'md-arrow-down'}
              style={{color: Colors.tintColor, paddingTop: 3}}
            />
            <Text style={{color: Colors.tintColor}}> {this.state.value}</Text>
          </View>
        )}
        actions={[
          ...Array(this.props.options.length).fill().map((_, idx) => () => this.itemPressed(this.props.options[idx])),
          _ => null
        ]} />
    );
  }
}
