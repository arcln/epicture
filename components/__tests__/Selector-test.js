import {View} from 'react-native';
import React from 'react';
import Selector from '../Selector';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <View style={{flexDirection: 'row', justifyContent: 'flex-end', padding: 10}}>
      <Selector
        style={{marginRight: 10}}
        options={['Most recent', 'Viral', 'Top']}
        default='Most recent'
        onChange={value => null}
      />
      <Selector
        options={['Today', 'This week', 'This month', 'This year', 'All time']}
        default='All time'
        onChange={value => null}
      />
    </View>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
