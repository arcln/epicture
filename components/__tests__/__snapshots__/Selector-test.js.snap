// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "flex-end",
      "padding": 10,
    }
  }
>
  <View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "marginRight": 10,
          }
        }
      >
        <Text />
        <Text
          style={
            Object {
              "color": "#2f95dc",
            }
          }
        >
           
          Most recent
        </Text>
      </View>
    </View>
  </View>
  <View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Text />
        <Text
          style={
            Object {
              "color": "#2f95dc",
            }
          }
        >
           
          All time
        </Text>
      </View>
    </View>
  </View>
</View>
`;
